File: Selenium/Activities/Activity11.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

with webdriver.Firefox() as driver:
    wait = WebDriverWait(driver, timeout=10)
    driver.get("https://training-support.net/webelements/dynamic-controls")
    title=driver.title
    print("Title of the page is: ",title)

    checkbox=driver.find_element(By.ID,"checkbox")
    print("Is Checkbox visible: ",checkbox.is_displayed())

    driver.find_element(By.XPATH,"//button[text()='Toggle Checkbox']").click()
    wait.until(EC.invisibility_of_element(checkbox))

    driver.find_element(By.XPATH,"//button[text()='Toggle Checkbox']").click()
    wait.until(EC.element_to_be_clickable(checkbox)).click()

    print("Is Checkbox selected: ",checkbox.is_selected())

    

File: Selenium/Activities/Activity11.java
import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activity11 {

	public static void main(String[] args) {
		
        WebDriver driver=new FirefoxDriver();
		
        driver.get("https://training-support.net/webelements/dynamic-controls");
		
		String title=driver.getTitle();
		System.out.println("Title of the page is: "+title);

		
		WebElement checkbox=driver.findElement(By.id("checkbox"));
		System.out.println("Is Checkbox visible: "+checkbox.isDisplayed());
		
		driver.findElement(By.xpath("//button[text()='Toggle Checkbox']")).click();
		
		WebDriverWait wait=new WebDriverWait(driver,Duration.ofSeconds(10));
		wait.until(ExpectedConditions.invisibilityOf(checkbox));
		
		driver.findElement(By.xpath("//button[text()='Toggle Checkbox']")).click();    	
		
		WebDriverWait wait1=new WebDriverWait(driver,Duration.ofSeconds(10));
		wait1.until(ExpectedConditions.elementToBeClickable(checkbox)).click();
		
		System.out.println("Is Checkbox selected: "+checkbox.isSelected());
		
		driver.quit();
		

	}

}
